src/Strategy/Scala/SbtDependencyTree.hs

```diff
@@ -0,0 +1,235 @@
+import Text.Megaparsec (
+  MonadParsec (eof, takeWhileP, try),
+  Parsec,
+  chunk,
+  empty,
+  errorBundlePretty,
+  many,
+  optional,
+  runParser,
+  sepBy,
+  some,
+  (<|>),
+ )
+import Text.Megaparsec.Char (alphaNumChar, char, eol)
+import Text.Megaparsec.Char.Lexer qualified as Lexer
+
+-- | Sbt Dependency Ascii tree task
+-- This only works with sbt v1.4.0 greater, or with sbt which has DependencyTreePlugin.
+-- Ref: https://www.scala-sbt.org/1.x/docs/sbt-1.4-Release-Notes.html#sbt-dependency-graph+is+in-sourced
+sbtDepTreeCmd :: Command
+sbtDepTreeCmd =
+  Command
+    { cmdName = "sbt"
+    , cmdArgs =
+        [ "--batch" -- ensure sbt does not enter repl mode!
+        , "--no-colors"
+        , "dependencyTree"
+        ]
+    , cmdAllowErr = Never
+    }
+
+data SbtDep = SbtDep
+  { artifact :: SbtArtifact
+  , requires :: [SbtDep]
+  }
+  deriving (Eq, Ord, Show)
+
+type Parser = Parsec Void Text
+
+sc :: Parser ()
+sc = Lexer.space (void $ some $ char ' ' <|> char '\t') empty empty
+
+lexeme :: Parser a -> Parser a
+lexeme = Lexer.lexeme sc
+
+symbol :: Text -> Parser Text
+symbol = Lexer.symbol sc
+
+-- | Parses package identifier.
+--
+-- @Examples:
+-- >> parseTest parseValidProjectIdentifier "org.a.b" = "org.a.b"
+-- >> parseTest parseValidProjectIdentifier "org.a.b-c" = "org.a.b-c"
+-- >> parseTest parseValidProjectIdentifier "org.a.b:" = "org.a.b"
+-- >> parseTest parseValidProjectIdentifier "org.a.b " = "org.a.b"
+parseValidProjectIdentifier :: Parser Text
+parseValidProjectIdentifier = toText <$> some (alphaNumChar <|> char '.' <|> char '-' <|> char '_')
+
+-- | Parses Sbt Artifact
+--
+-- @Example:
+-- >> parseTest parseSbtArtifact "com.typesafe:config:1.3.1" = SbtArtifact "com.typesafe" "config" "1.3.1"
+-- >> parseTest parseSbtArtifact "com.typesafe:config:1.3.1 [S]" = SbtArtifact "com.typesafe" "config" "1.3.1"
+parseSbtArtifact :: Parser SbtArtifact
+parseSbtArtifact =
+  SbtArtifact
+    <$> parseValidProjectIdentifier
+    <*> (":" >> parseValidProjectIdentifier)
+    <*> (":" >> parseValidProjectIdentifier)
+
+-- | Parses chars used for ascii graphing layout.
+parseAsciiGraphLayoutChars :: Parser Text
+parseAsciiGraphLayoutChars =
+  toText
+    <$> many
+      ( char ' '
+          <|> char '|'
+          <|> char '+'
+          <|> char '-'
+          <|> char '#'
+      )
+
+-- | Parses evicted version identifier
+-- >> parseEviction "(evicted by: 2.0)" = "2.0"
+parseEviction :: Parser (Text)
+parseEviction = symbol "(evicted by:" *> parseValidProjectIdentifier <* symbol ")"
+
+-- | Parses ASCII tree
+--
+-- parent
+--   +-childA
+--   +-childB
+--   |   +-childC
+--   |
+--   +-childD
+--
+-- Reference: https://github.com/sbt/sbt/blob/develop/main/src/main/scala/sbt/internal/SettingGraph.scala#L84
+sbtTreeParser :: Parser [SbtDep]
+sbtTreeParser = concat <$> ((try (parseDeps 0) <|> ignoredLine) `sepBy` eol) <* eof
+  where
+    isEndLine :: Char -> Bool
+    isEndLine '\n' = True
+    isEndLine '\r' = True
```

Comment:
Since you are handling line endings explicitly, should we have a test of `sbtTreeParser` with the windows `\r\n` line ending? I couldn't get current tests to fail with that line-ending, but it might be good to document that we expect/can handle windows line endings in one of the test strings.